chore: support configuring `pmtiles.{skip_credentials,force_path_style}` (#1848)
urbica/martin: docs/src/env-vars.md

## Environment Variables

> You can also configure Martin using environment variables, but only if the configuration file is not used.

Reviewer: ```suggestion
You can configure Martin using environment variables, but only if the configuration file is not used. The configuration file itself may use environment variables if needed.
```

Author: may?
I think it can do that or are there cases where we don't?

```suggestion
You can configure Martin using environment variables, but only if the configuration file is not used.
The configuration file itself can use environment variables if needed.
```

| `PGSSLKEY` <br/> `ssl_key`               | `./postgresql.key`                          | A file with the key for the client SSL certificate. [docs](https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-CONNECT-SSLKEY)                                                                |
| `PGSSLROOTCERT` <br/> `ssl_root_cert`    | `./root.crt`                                | A file with trusted root certificate(s). The file should contain a sequence of PEM-formatted CA certificates. [docs](https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-CONNECT-SSLROOTCERT) |
| `AWS_LAMBDA_RUNTIME_API`                 |                                             | If defined, connect to AWS Lambda to handle requests. The regular HTTP server is not used. See [Running in AWS Lambda](run-with-lambda.md)                                                                 |

> We also have the following [pmtiles via S3 specific configuration options](sources-files.md#serving-pmtiles-via-s3):

Reviewer: ```suggestion
[PMTiles S3 access](sources-files.md#serving-pmtiles-via-s3) also supports these configuration options:
```

Author: ```suggestion
To [access PMTiles via S3](sources-files.md#serving-pmtiles-via-s3) also supports these configuration options:
```

> - `AWS_ACCESS_KEY_ID`
> - `AWS_SECRET_ACCESS_KEY`
> - `AWS_SESSION_TOKEN`
> - `AWS_PROFILE`
> - `AWS_REGION`
> - `AWS_NO_CREDENTIALS`
> - `AWS_S3_FORCE_PATH_STYLE`

Author: ```suggestion
| Environment var <br/> Config File key                      | Example                                    | Description                                                                                                                                                               |
| ---------------------------------------------------------- | ------------------------------------------ | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| `AWS_ACCESS_KEY_ID` <br/> `-`                              | `AKIAEXAMPLE12345678`                      | AWS access key ID used for authenticating requests when using long-term or temporary credentials.                                                                         |
| `AWS_SECRET_ACCESS_KEY` <br/> `-`                          | `wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY` | AWS secret access key paired with the access key ID.                                                                                                                      |
| `AWS_SESSION_TOKEN` <br/> `-`                              | `FwoGZXIvYXdzEF0aD...`                     | Session token used with temporary security credentials (e.g., from AWS STS). Required if you're using `AssumeRole`.                                                       |
| `AWS_PROFILE` <br/> `-`                                    | `default`                                  | Specifies which named profile to use from the AWS credentials/config files.                                                                                               |
| `AWS_REGION` <br/> `-`                                     | `us-west-2`                                | Sets the AWS region to send requests to, e.g., `us-east-1`, `eu-central-1`.                                                                                               |
| `AWS_NO_CREDENTIALS` <br/> `pmtiles.no_credentials`        | `true`                                     | Disable credential loading and to send requests anonymously for publicly available buckets.                                                                               |
| `AWS_S3_FORCE_PATH_STYLE` <br/> `pmtiles.force_path_style` | `true`                                     | Forces the AWS SDK to use path-style URLs for S3 like `s3.amazonaws.com/bucket/key`) instead of virtual-hosted style. Useful for local S3-compatible services like MinIO. |
```

> - `AWS_S3_FORCE_PATH_STYLE`

Author: main question is likely for this PR if these env vars are fine.

Reviewer: i think these AWS env vars are pretty standard, and I think they are ok to use.

Reviewer: P.S. Let's document these the same way as PG env vars, together with the corresponding config keys. See above.

Author: > together with the corresponding config keys

They currently don't have config associated with them.
Should they be added under an `pmtiles.aws.*` or `plmtiles.aws_*`?

| `PGSSLKEY` <br/> `ssl_key`               | `./postgresql.key`                          | A file with the key for the client SSL certificate. [docs](https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-CONNECT-SSLKEY)                                                                |
| `PGSSLROOTCERT` <br/> `ssl_root_cert`    | `./root.crt`                                | A file with trusted root certificate(s). The file should contain a sequence of PEM-formatted CA certificates. [docs](https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-CONNECT-SSLROOTCERT) |

> | `AWS_LAMBDA_RUNTIME_API`                 |                                             | If defined, connect to AWS Lambda to handle requests. The regular HTTP server is not used. See [Running in AWS Lambda](run-with-lambda.md)                                                                 |
> 
> To [access PMTiles via S3](sources-files.md#serving-pmtiles-via-s3) also supports these configuration options:
> 
> | Environment var <br/> Config File key                      | Example                                    | Description                                                                                                                                                               |
> | ---------------------------------------------------------- | ------------------------------------------ | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
> | `AWS_ACCESS_KEY_ID` <br/> `-`                              | `AKIAEXAMPLE12345678`                      | AWS access key ID used for authenticating requests when using long-term or temporary credentials.                                                                         |
> | `AWS_SECRET_ACCESS_KEY` <br/> `-`                          | `wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY` | AWS secret access key paired with the access key ID.                                                                                                                      |
> | `AWS_SESSION_TOKEN` <br/> `-`                              | `FwoGZXIvYXdzEF0aD...`                     | Session token used with temporary security credentials (e.g., from AWS STS). Required if you're using `AssumeRole`.                                                       |
> | `AWS_PROFILE` <br/> `-`                                    | `default`                                  | Specifies which named profile to use from the AWS credentials/config files.                                                                                               |
> | `AWS_REGION` <br/> `-`                                     | `us-west-2`                                | Sets the AWS region to send requests to, e.g., `us-east-1`, `eu-central-1`.                                                                                               |

Author: ```suggestion
| `AWS_LAMBDA_RUNTIME_API` <br/> -         |                                             | If defined, connect to AWS Lambda to handle requests. The regular HTTP server is not used. See [Running in AWS Lambda](run-with-lambda.md)                                                                 |

To [access PMTiles via S3](sources-files.md#serving-pmtiles-via-s3) also supports these configuration options:

| Environment var <br/> Config File key                      | Example                                    | Description                                                                                                                                                               |
| ---------------------------------------------------------- | ------------------------------------------ | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| `AWS_ACCESS_KEY_ID` <br/> -                                | `AKIAEXAMPLE12345678`                      | AWS access key ID used for authenticating requests when using long-term or temporary credentials.                                                                         |
| `AWS_SECRET_ACCESS_KEY` <br/> -                            | `wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY` | AWS secret access key paired with the access key ID.                                                                                                                      |
| `AWS_SESSION_TOKEN` <br/> -                                | `FwoGZXIvYXdzEF0aD...`                     | Session token used with temporary security credentials (e.g., from AWS STS). Required if you're using `AssumeRole`.                                                       |
| `AWS_PROFILE` <br/> -                                      | `default`                                  | Specifies which named profile to use from the AWS credentials/config files.                                                                                               |
| `AWS_REGION` <br/> -                                       | `us-west-2`                                | Sets the AWS region to send requests to, e.g., `us-east-1`, `eu-central-1`.                                                                                               |
```
